src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py

```diff
@@ -48,29 +48,20 @@ def compress(self):
         """
         Compress the model, register a hook for collecting activations.
         """
-        if self.modules_wrapper is not None:
-            # already compressed
-            return self.bound_model
-        else:
-            self.modules_wrapper = []
-        modules_to_compress = self.detect_modules_to_compress()
-        for layer, config in modules_to_compress:
-            wrapper = self._wrap_modules(layer, config)
-            self.modules_wrapper.append(wrapper)
-            self.collected_activation[layer.name] = []
-
-            def _hook(module_, input_, output, name=layer.name):
+        for wrapper in self.get_modules_wrapper():
+            self.collected_activation[wrapper.name] = []
+
+            def _hook(module_, input_, output, name=wrapper.name):
                 if len(self.collected_activation[name]) < self.statistics_batch_num:
                     self.collected_activation[name].append(self.activation(output.detach().cpu()))
```

Comment:
is there data race here?

Reply:
I mean DataParallel

Reply:
There is. I haven't come up with how to remove the data racing here yet.